scripts/irods/test/test_resource_types.py

```diff
@@ -2385,6 +2385,142 @@ def test_auto_repl_on__2941(self):
+            self.admin.assert_icommand("iadmin modrepl logical_path " + logical_path + " replica_number 0 DATA_REPL_STATUS 0")
+            self.admin.assert_icommand("iadmin modrepl logical_path " + logical_path + " replica_number 1 DATA_REPL_STATUS 1")
+
+            # get the stale replica
+            get_filename = "getfile.txt"
+            self.admin.assert_icommand("iget -n 0 " + filename + " " +  get_filename)
+
+            # assert on the stale replica file size
+            self.assertEqual(os.path.getsize(get_filename), 20)
+
+            # get the file without selecting the replica
+            self.admin.assert_icommand("iget -f " + filename + " " +  get_filename)
+
+            # assert on the good replica file size
+            self.assertEqual(os.path.getsize(get_filename), 10)
+
+        finally:
+            # cleanup
+            self.admin.assert_icommand("irm -f " + filename)
+            lib.remove_file_if_exists(filename)
+            lib.remove_file_if_exists(stale_replica_filename)
+            lib.remove_file_if_exists(get_filename)
+
+    def test_direct_request_for_cache_archive_replicas__issue_6926(self):
+
+        try:
+            filename = "directrequestcachearchivereplica.txt"
+            get_filename = "getfile.txt"
+            lib.make_file(filename, 10, 'arbitrary')
+
+            logical_path = os.path.join(self.admin.session_collection, filename)
+
+            self.admin.assert_icommand("iput " + filename)
+
+            # attempt to get the replica directly from archive - should fail
+            self.admin.assert_icommand(['iget', '-n1', filename, get_filename], 'STDERR', 'DIRECT_CHILD_ACCESS')
+
+            # attempt to get the replica directly from cache - should pass
+            self.admin.assert_icommand(['iget', '-n0', filename, get_filename])
+
+            # assert that the retrieved file is the right size
+            self.assertEqual(os.path.getsize(get_filename), 10)
+            os.remove(get_filename)
+
+            # get the file without requesting replica or resource
+            self.admin.assert_icommand(['iget', filename, get_filename])
+
+            # assert that the retrieved file is the right size
+            self.assertEqual(os.path.getsize(get_filename), 10)
+            os.remove(get_filename)
+
+            # get the file specifying the compound resource - should pass
+            self.admin.assert_icommand(['iget', '-R', 'demoResc', filename, get_filename])
+
+            # assert that the retrieved file is the right size
+            self.assertEqual(os.path.getsize(get_filename), 10)
+            os.remove(get_filename)
+
+        finally:
+            # cleanup
+            self.admin.assert_icommand("irm -f " + filename)
+            lib.remove_file_if_exists(filename)
+            lib.remove_file_if_exists(get_filename)
+
+    def test_direct_request_for_cache_archive_replicas_when_policy_is_set_to_prefer_archive__issue_6926(self):
```

Comment:
Should we port `test_direct_request_for_cache_archive_replicas__issue_6926` (and maybe the other test) over to the `Test_Resource_ReplicationToTwoCompoundResourcesWithPreferArchive` suite? Maybe just adding this test to this suite (as you are doing here) is enough.

Reply:
I didn't realize that that suite existed.  I think it makes sense to just copy this over to that one and remove the specific prefer archive test.  Agree?

Reply:
On second thought, that class is very specific with a replication resource above two archive resources with preferred archive enabled.  I am not sure what the point of that is but I am inclined to leave this as is.

Reply:
Yeah, I am not married to the idea, just wanted to make it known that we have a test suite for it (sorta). This is fine as is.